src/screens/WhotoFollow/index.js

```diff
@@ -173,65 +173,65 @@ const WhotoFollow = () => {
-          setToken(res.token);
-          try {
-            const userId = await JwtDecode(res.token).user_id;
-            const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
-            setProfileId({
-              anonProfileId: anonymousUserId,
-              signedProfileId: userId,
-              token: res.token,
-              anonymousToken: res.anonymousToken
+      registerUser(data)
+        .then(async (res) => {
+          setFetchRegister(false);
+          if (res.code === 200) {
+            TokenStorage.set(res);
+            setToken(res.token);
+            try {
+              const userId = await JwtDecode(res.token).user_id;
+              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              setProfileId({
+                anonProfileId: anonymousUserId,
+                signedProfileId: userId,
+                token: res.token,
+                anonymousToken: res.anonymousToken
+              });
+            } catch (e) {
+              crashlytics().recordError(new Error(e));
+            }
+            showMessage({
+              message: 'Welcome to Better Social',
+              type: 'success',
+              backgroundColor: colors.holytosca
+            });
+            setTimeout(() => {
+              create();
+              setImage(null, usersDispatch);
+              setInitialValue({id: res.token});
+            }, 2000);
+          } else {
+            crashlytics().recordError(new Error(res));
+            showMessage({
+              message: 'Cannot connect to server, please try again later',
+              type: 'danger',
+              backgroundColor: colors.red
             });
-          } catch (e) {
-            crashlytics().recordError(new Error(e));
           }
-          showMessage({
-            message: 'Welcome to Better Social',
-            type: 'success',
-            backgroundColor: colors.holytosca
-          });
-          setTimeout(() => {
-            create();
-            setImage(null, usersDispatch);
-            setInitialValue({id: res.token});
-          }, 2000);
-        } else {
-          crashlytics().recordError(new Error(res));
+        })
+        .catch((error) => {
+          crashlytics().recordError(new Error(error.response));
+          setFetchRegister(false);
           showMessage({
             message: 'Cannot connect to server, please try again later',
             type: 'danger',
             backgroundColor: colors.red
           });
-        }
-      })
-      .catch((error) => {
-        crashlytics().recordError(new Error(error.response));
-        setFetchRegister(false);
-        showMessage({
-          message: 'Cannot connect to server, please try again later',
-          type: 'danger',
-          backgroundColor: colors.red
         });
-      });
+    } catch (e) {
+      console.error('Error during registration:', e);
+      // Handle the error appropriately here
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this code could be improved. Currently, if an error occurs during the image upload process (lines 179-183), it is caught and logged (line 232), but the registration process continues regardless (lines 185-230). This could lead to unexpected behavior if the image upload fails but the user registration succeeds. It would be better to stop the registration process if the image upload fails and inform the user about the issue.

```diff
    try {
      const profilePic = usersState?.photoUrl;
      if (profilePic && profilePic !== DEFAULT_PROFILE_PIC_PATH) {
         const uploadedImageUrl = await ImageUtils.uploadImageWithoutAuth(
           data?.users?.profile_pic_path
         );
         data.users.profile_pic_path = uploadedImageUrl?.data?.url;
       }
 
      registerUser(data)
...
    } catch (e) {
      console.error('Error during registration:', e);
      // Handle the error appropriately here
+       showMessage({
+         message: 'An error occurred during registration. Please try again.',
+         type: 'danger',
+         backgroundColor: colors.red
+       });
+       setFetchRegister(false);
    }
```

Also, consider using a more specific error message when catching errors from `registerUser` (lines 222-229). The current message "Cannot connect to server, please try again later" may not accurately describe the problem if the error is not related to server connectivity.

```diff
        .catch((error) => {
          crashlytics().recordError(new Error(error.response));
          setFetchRegister(false);
           showMessage({
-              message: 'Cannot connect to server, please try again later',
+              message: `Registration failed: ${error.message}`,
             type: 'danger',
             backgroundColor: colors.red
           });
         });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->